services/inputhost/inputhost.go

```diff
@@ -864,13 +864,26 @@ func (h *InputHost) Report(reporter common.LoadReporter) {
 		NumberOfConnections:     common.Int64Ptr(h.hostMetrics.Get(load.HostMetricNumOpenConns)),
 		IncomingMessagesCounter: common.Int64Ptr(msgsInPerSec),
 		IncomingBytesCounter:    common.Int64Ptr(bytesInSinceLastReport),
+		NodeStatus:              common.NodeStatusPtr(h.GetNodeStatus()),
 	}
 
 	reporter.ReportHostMetric(hostMetrics)
+	return *(hostMetrics.NumberOfConnections)
+}
+
+// Report is the implementation for reporting host specific load to controller
+func (h *InputHost) Report(reporter common.LoadReporter) {
+
+	now := time.Now().UnixNano()
+	intervalSecs := (now - h.lastLoadReportedTime) / int64(time.Second)
+	if intervalSecs < 1 {
+		return
+	}
+
+	numConns := h.reportHostMetric(reporter, intervalSecs)
 	h.lastLoadReportedTime = now
 
 	// Also update the metrics reporter to make sure the connection gauge is incremented
-	numConns := *(hostMetrics.NumberOfConnections)
 	h.m3Client.UpdateGauge(metrics.PubConnectionStreamScope, metrics.InputhostPubConnection, numConns)
```

Comment:
i didn't want to do m3 call when we were going down.. that's the only reason for not moving this..